homeassistant/components/squeezebox_player/media_player.py

```diff
@@ -0,0 +1,223 @@
+                "configuration_url"
+            ] = f"http://{self.player.device_address}"
+        self.update_attributes()
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        self.update_attributes()
+        self.async_on_remove(
+            self.slimserver.subscribe(
+                self._on_slim_event,
+                (
+                    EventType.PLAYER_UPDATED,
+                    EventType.PLAYER_CONNECTED,
+                    EventType.PLAYER_DISCONNECTED,
+                    EventType.PLAYER_NAME_RECEIVED,
+                    EventType.PLAYER_RPC_EVENT,
+                ),
+                player_filter=self.player.player_id,
+            )
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return availability of entity."""
+        return self.player.connected
+
+    @property
+    def state(self) -> str:
+        """Return current state."""
+        if not self.player.powered:
+            return STATE_OFF
+        return STATE_MAPPING[self.player.state]
+
+    @callback
+    def update_attributes(self) -> None:
+        """Handle player updates."""
+        self._attr_name = self.player.name
+        self._attr_volume_level = self.player.volume_level / 100
+        self._attr_media_position = self.player.elapsed_seconds
+        self._attr_media_position_updated_at = utcnow()
+        self._attr_media_content_id = self.player.current_url
+        self._attr_media_content_type = "music"
+
+    async def async_media_play(self) -> None:
+        """Send play command to device."""
+        await self.player.play()
+
+    async def async_media_pause(self) -> None:
+        """Send pause command to device."""
+        await self.player.pause()
+
+    async def async_media_stop(self) -> None:
+        """Send stop command to device."""
+        await self.player.stop()
+
+    async def async_set_volume_level(self, volume: float) -> None:
+        """Send new volume_level to device."""
+        volume = int(volume * 100)
+        await self.player.volume_set(volume)
+
+    async def async_mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        await self.player.mute(mute)
+
+    async def async_turn_on(self) -> None:
+        """Turn on device."""
+        await self.player.power(True)
+
+    async def async_turn_off(self) -> None:
+        """Turn off device."""
+        await self.player.power(False)
+
+    async def async_play_media(self, media_type: str, media_id: str, **kwargs) -> None:
+        """Send the play_media command to the media player."""
+        # Handle media_source
+        if media_source.is_media_source_id(media_id):
+            sourced_media = await media_source.async_resolve_media(self.hass, media_id)
+            media_id = sourced_media.url
+            media_type = sourced_media.mime_type
+
+        if not media_type.startswith("audio/"):
+            media_type = None  # type: ignore[assignment]
```

Comment:
We should be able to avoid the ignore if we pretype the variable as optional at the beginning of the method body.